govcd/vdc.go

```diff
@@ -125,6 +133,10 @@ func (vdc *Vdc) Refresh() error {
 
 func (vdc *Vdc) FindVDCNetwork(network string) (OrgVDCNetwork, error) {
 
+	err := vdc.Refresh()
```

Comment:
We have discussed this addition in Issue #105 (and in a recent meeting).
If we don't refresh, we get stale results, especially when we look for a network that was just created.
Either we do this refresh, or we should update the VDC object in memory after a creation. This kind of refresh is performed in other entities as well, and I assume it's done for the same reason.
In short, without a refresh we are using a VDC cache, without a way of knowing when the cache was invalidated. We can fix it, but it is not to be quick or pretty.

Reply:
Any automatic solution to this needs to be validated by realistic performance tests.  We're adding what look like expensive network calls.  (I'm going to check into how we handle this issue for applications like the vCD UI, which presumably has this concern in spaces.)

Reply:
This is the way it is done currently throughout the library:
- In various vApp functions vdc.Refresh() is done;
- In various network functions vdc.Refresh() is done;
- In various firewall and VPN functions edgeGateway.Refresh() is done.

Given the above, let's do the following:
1.) Accept this vdc.Refresh() in FindVDCNetwork(...) function as well, as it makes it is both consistent with current model and, more importantly, easier for users to program with the lib.
2.) If we see a performance issue down the road, we fix it as such by adding a possibility to tune the behaviour of such functions by asking _not_ refresh explicitly.

Finally, there's not much value in making this refresh smart (by tracking the state of created resources), because new resources may be easily created _outside_ the library in:
a.) vCD GUI.
b.) Outside processes, like Terraform.
c.) Python CLI, etc.

Reply:
This is the way it is done currently throughout the library:
- In various vApp functions vdc.Refresh() is done;
- In various network functions vdc.Refresh() is done;
- In various firewall and VPN functions edgeGateway.Refresh() is done.

Given the above, let's do the following:
1.) Accept this vdc.Refresh() in FindVDCNetwork(...) function as well, as it makes it is both consistent with current model and, more importantly, easier for users to program with the lib.
2.) If we see a performance issue down the road, we fix it as such by adding a possibility to tune the behaviour of such functions by asking _not_ refresh explicitly.

Finally, there's not much value in making this refresh smart (by tracking the state of created resources), because new resources may be easily created _outside_ the library in:
a.) vCD GUI.
b.) Outside processes, like Terraform.
c.) Python CLI, etc.

Reply:
This is the way it is done currently throughout the library:
- In various vApp functions vdc.Refresh() is done;
- In various network functions vdc.Refresh() is done;
- In various firewall and VPN functions edgeGateway.Refresh() is done.

Given the above, let's do the following:
1.) Accept this vdc.Refresh() in FindVDCNetwork(...) function as well, as it makes it is both consistent with current model and, more importantly, easier for users to program with the lib.
2.) If we see a performance issue down the road, we fix it as such by adding a possibility to tune the behaviour of such functions by asking _not_ refresh explicitly.

Finally, there's not much value in making this refresh smart (by tracking the state of created resources), because new resources may be easily created _outside_ the library in:
a.) vCD GUI.
b.) Outside processes, like Terraform.
c.) Python CLI, etc.